sdk/core/Azure.Core/api/Azure.Core.netcoreapp2.1.cs

```diff
@@ -1134,6 +1134,12 @@ public abstract partial class ObjectSerializer
         public abstract System.Threading.Tasks.ValueTask SerializeAsync(System.IO.Stream stream, object? value, System.Type inputType, System.Threading.CancellationToken cancellationToken);
         public virtual System.Threading.Tasks.ValueTask<System.BinaryData> SerializeAsync(object? value, System.Type? inputType = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public abstract partial class SchemaValidator<T>
+    {
+        protected SchemaValidator() { }
+        public abstract T GenerateSchema(System.Type dataType);
```

Comment:
I don't believe the serializer would be able to support a scenario without generation. We need to have a schema in text format to be able to validate with the service that the schema exists in the schema registry. Because of the limitations of JSON schema (all the different draft formats) and the lack of any good third-party libraries, we have no way to get the schema from a type ourselves.

From a customer perspective, an alternative could be to return the string form of the schema in this method from some kind of local storage. For example, if an application knows they only use one schema, they might just hard code this method to return that one string every time.